packages/back-end/src/models/GeneratedHypothesis.ts

```diff
@@ -0,0 +1,264 @@
+  if (!process.env.SUPABASE_URL || !process.env.SUPABASE_SERVICE_KEY)
+    throw new Error("Supabase keys missing");
+
+  const supabase = createClient(
+    process.env.SUPABASE_URL,
+    process.env.SUPABASE_SERVICE_KEY
+  );
+
+  const { data: siteHypothesis, error: siteHypErr } = await supabase
+    .from("sites_hypotheses")
+    .select("*, sites(url), sites_dom_mutations(sdk_payload)")
+    .eq("uuid", uuid)
+    .single();
+
+  if (siteHypErr) throw new Error(siteHypErr.message);
+
+  const {
+    hypothesis: { slug, oneLineSummary, hypothesis, control, variant },
+    sites: { url },
+    sites_dom_mutations,
+  } = siteHypothesis;
+  const { sdk_payload } = sites_dom_mutations || {};
+
+  const experimentToCreate: Pick<
+    ExperimentInterface,
+    | "name"
+    | "owner"
+    | "description"
+    | "hypothesis"
+    | "hashAttribute"
+    | "status"
+    | "trackingKey"
+    | "variations"
+    | "phases"
+  > = {
+    name: slug,
+    owner: userId,
+    description: oneLineSummary,
+    hypothesis,
+    hashAttribute: "id",
+    status: "draft",
+    trackingKey: slug,
+    variations: [
+      {
+        name: "Control",
+        description: control,
+        key: "0",
+        screenshots: [],
+        id: uniqid("var_"),
+      },
+      {
+        name: `Variation 1`,
+        description: variant,
+        key: "1",
+        screenshots: [],
+        id: uniqid("var_"),
+      },
+    ],
+    phases: [
+      {
+        coverage: 1,
+        dateStarted: new Date(),
+        dateEnded: new Date(),
+        name: "Main",
+        reason: "",
+        variationWeights: [0.5, 0.5],
+        condition: "",
+        namespace: { enabled: false, name: "", range: [0, 1] },
+      },
+    ],
+  };
+
+  const createdExperiment = await createExperiment({
+    data: experimentToCreate,
+    context,
+  });
+
+  await upsertWatch({
+    userId,
+    organization: org.id,
+    item: createdExperiment.id,
+    type: "experiments",
+  });
+
+  // create feature flag or visual change
+  if (sdk_payload) {
+    // visual change
+    await createVisualChangeset({
+      experiment: createdExperiment,
+      context,
+      urlPatterns: sdk_payload.urlPatterns,
+      editorUrl: sdk_payload.targetUrl,
+      visualChanges: [
+        {
+          description: "",
+          id: uniqid("vc_"),
+          css: "",
+          js: "",
+          domMutations: sdk_payload.variations[0].domMutations,
+          variation: createdExperiment.variations[0].id,
+        },
+        {
+          description: "",
+          id: uniqid("vc_"),
+          css: "",
+          js: "",
+          domMutations: sdk_payload.variations[1].domMutations,
+          variation: createdExperiment.variations[1].id,
+        },
+      ],
+    });
+    // TODO enable pro trial if they are not already enabled
+  } else {
+    // linked feature flag
+    const featureId = `${slug}-feature-flag`;
+    await createFeature(context, {
```

Comment:
do we need to do this? won't it just run as a standalone exp?

Reply:
no, you can't run an exp without either a FF, visual change, or URL redirect. it's part of what we discussed is necessary for a 'pregenerated experiment'. creating this gives us one less step towards a running exp.

Reply:
got it, ok